### PR TITLE
DCES-1119: Added test coverage to using empty target files and IO Exception thrown

### PR DESCRIPTION
**Pull request by tanovellino**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LASB-XXX)

Just added more tests to increase test coverage

## Checklist

Before you ask people to review this PR:

- [x] Tests should be passing: `./gradlew test`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/report/service/CSVFileService.java [hunk 1]
```diff
@@ -35,7 +35,7 @@ public class CSVFileService {
 
     private static final String TEMPLATE_TITLE = "%s REPORTING DATE FROM: %s | REPORTING DATE TO: %s | REPORTING PRODUCED ON: %s" + System.lineSeparator();
 
-    private static final String FDC_HEADER = "MAAT ID, Sentence Date, Calculation Date, Final Cost, LGFS Cost, AGFS COST, Transmission Date" + System.lineSeparator();
+    public static final String FDC_HEADER = "MAAT ID, Sentence Date, Calculation Date, Final Cost, LGFS Cost, AGFS COST, Transmission Date" + System.lineSeparator();
```
Comment (LAA-Dave-Edwards) [inline dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/report/service/CSVFileService.java:38]: Why is this moved over to public?

File: dces-report-service/src/test/java/uk/gov/justice/laa/crime/dces/report/service/CSVFileServiceTest.java [hunk 2]
```diff
@@ -17,76 +18,155 @@
...
+        softly.assertThat(output).contains(contFile.get(0).getMaatId());
+    }
+
+    @Test
+    void GivenFileNotWriteable_WhenCallingWriteContributionToCsv_thenShouldThrowIOException() throws IOException {
+        List<ContributionCSVDataLine> contFile = List.of();
+        LocalDate date = LocalDate.now();
+        testFile = File.createTempFile("test", ".csv");
+
+        if (testFile.setWritable(false)) { // This will force IO Exception
```
Comment (LAA-Dave-Edwards) [inline dces-report-service/src/test/java/uk/gov/justice/laa/crime/dces/report/service/CSVFileServiceTest.java:100]: Good commenting 👍

**Review by LAA-Dave-Edwards — APPROVED** [review]
All good apart from the public change.

File: dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/report/service/CSVFileService.java (hunk 1 above)
Comment (tanovellino) [reply dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/report/service/CSVFileService.java:38]: I wanted to be available for testing
